### DISCUSSION TITLE
Can i use npm package that has third party sdk in android - ios side with react-native?

### DISCUSSION THREAD

**Discussion by ShumGG** [npm/feedback#814] [General]
building my npm package for my react-native app because i need to use native modules to show some ads, the company sent me their sdk so i can build it, im using the command from the page to start my package: **npx create-react-native-library react-native-awesome-module**, put the aar file for android and the .framework folder for ios, testing the package locally it works and can use the sdk.

Problem now is that i publish to npm with the sdk, i install it in my app and i need to build a development client using eas to test the native modules, when building for ios im getting the error: 

```
 (node_modules/react-native-innity-remaster/ios/EngageAd.swift:10:8)

   8 | import Foundation
   9 | 
> 10 | import IMAd;
     |        ^ no such module 'IMAd'
  11 | 
  12 | class EngageAD: UIViewController {
           ......
  }
```
 
looks like when building with eas it can detect the .framework i put in the project?.

So i dont know if i cant publish npm package that uses third party sdk with it or its just impossible

**Comment by ljharb** [comment]
This seems like a question for eas, or something in the swift ecosystem, and not npm?
